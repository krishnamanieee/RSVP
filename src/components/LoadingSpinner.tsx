import React from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  View,
  Text,
  Platform,
} from 'react-native';
import Theme from '../theme';

interface IProps {
  visible: boolean;
  message?: string;
  size?: 'small' | 'large';
}

class LoadingSpinner extends React.PureComponent<IProps, {}> {
  public static defaultProps = {
    visible: false,
    message: 'Please Wait..',
  };

  public render() {
    return this.renderSpinner();
  }

  public renderSpinner() {
    const {visible, size} = this.props;
    if (!visible) {
      return null;
    }

    return (
      <View style={styles.container} key={`spinner_${Date.now()}`}>
        <View style={styles.background}>
          <View style={styles.contentContainer}>
            <ActivityIndicator
              color={Theme.colors.primaryColor}
              size={size ? size : 'small'}
            />
            {Platform.OS === 'android' && (
              <Text style={styles.textStyle}>Please Wait...</Text>
            )}
          </View>
        </View>
      </View>
    );
  }
}

export default LoadingSpinner;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Theme.colors.transparentGray,
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
  },
  background: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentContainer: {
    backgroundColor: Theme.colors.white,
    padding: 20,
    borderRadius: Platform.OS === 'android' ? 0 : 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textStyle: {
    color: Theme.colors.black,
    marginHorizontal: 12,
  },
});
