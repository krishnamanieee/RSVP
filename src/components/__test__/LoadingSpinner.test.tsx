import React from 'react';
import renderer from 'react-test-renderer';
import LoadingSpinner, {ILoadingSpinnerProps} from '../LoadingSpinner';
let props: ILoadingSpinnerProps = {
  message: '',
  size: 'small',
  visible: false,
};
describe('LoadingSpinner', () => {
  it('should match match snapshot', () => {
    const tree = renderer.create(<LoadingSpinner {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
