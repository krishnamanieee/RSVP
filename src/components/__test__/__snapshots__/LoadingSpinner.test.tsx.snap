// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should match match snapshot 1`] = `null`;
